File: api/src/main/java/org/pmiops/workbench/db/model/ParticipantIdAndCohortStatus.java
package org.pmiops.workbench.db.model;

import org.pmiops.workbench.model.CohortStatus;

/**
 * Projection from {@link ParticipantCohortStatus} that contains just the participant ID and
 * cohort status.
 */
public interface ParticipantIdAndCohortStatus {

  Key getParticipantKey();

  CohortStatus getStatus();

  public interface Key {
    public Long getParticipantId();
  }
}

File: api/src/main/java/org/pmiops/workbench/db/model/ParticipantIdAndCohortStatus.kt
package org.pmiops.workbench.db.model

import org.pmiops.workbench.model.CohortStatus

/**
 * Projection from [ParticipantCohortStatus] that contains just the participant ID and cohort
 * status.
 */
interface ParticipantIdAndCohortStatus {

    val participantKey: Key

    val status: CohortStatus

    interface Key {
        val participantId: Long?
    }
}
